internal/manage/devices/md.go

```diff
@@ -28,7 +28,7 @@ func (m *metadataCollector) add(d *gen.Device) *gen.DevicesMetadata {
 			m.seenFields[field] = seen
 			m.md.FieldCounts = append(m.md.FieldCounts, seen)
 		}
-		val, ok := getMessageString(field, d)
+		val, ok := compareMessageString(field, d, func(v string) bool { return true })
```

Comment:
Why shouldn't regex be used here btw? Regex is a lot more portable between languages when parsing values than string logic inside a loop.

Reply:
Mostly because regex is most useful when you're looking for a particular thing. What you're doing here is processing all input. The syntax of the path is simple enough that a loop over the runes of the string and a switch looking for `[` or `.` characters should be enough to do what you want. Additionally writing a parser (instead of finding patterns) will allow better error messages like `expecting ]` or `list index is not a number` which are hard to do with regex.

I'm not saying regex can't do it, but [as they say](https://blog.codinghorror.com/regular-expressions-now-you-have-two-problems/):
> Some people, when confronted with a problem, think "I know, I'll use regular expressions." Now they have two problems.